Book-A-Desk.Domain/Reservation.Command.BookADesk.fs

```diff
@@ -9,8 +9,11 @@ type BookADeskReservationCommand =
     }
 
 module BookADeskReservationCommand =
-    let provide validateCommand =
-        let execute (command:BookADesk) =
+    let provide getAllOffices =
+        let validate (command:BookADesk) reservationAggregate  =
+             BookADeskReservationValidator.validateCommand getAllOffices command reservationAggregate
```

Comment:
I moved the validation section again into the command because I think the validation should be visible here explicitly. What I meant with my original comment was that we can move the concrete validations into a separate module, but the orchestration of the validation should happen here. (We can refactor it in a separate PR)